### PR TITLE
feat: add record, autoStartMeeting, muteOnStart parameters to bbb

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
This PR add record, autoStartRecording, muteOnStart Features to BBB video-stream

require https://github.com/fossasia/open-event-server/pull/7835

### CODE REVIEW

**Comment by mariobehling** [comment]
Thank you. I did not see the feature start auto-recording in BBB yet. Would be definitely great to have it.

I think we can only merge this PR once we also have a place to access the recordings. Where would we access them?

**Comment by iamareebjamal** [comment]
In the edit page itself for now. We are in the process of implementing it https://github.com/fossasia/open-event-server/pull/7838

**Review by mariobehling — APPROVED** [review]

File: app/components/forms/events/view/videoroom-form.js [hunk 1]
```diff
@@ -115,6 +115,7 @@ export default class VideoroomForm extends Component.extend(FormMixin) {
 
   addBigBlueButton(channel) {
     this.data.stream.set('url', channel.get('url') + '/b/' + this.streamIdentifier);
+    this.data.stream.set('extra', { 'options': { 'record': true, 'autoStartRecording': false, 'muteOnStart': true } });
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.js:118]:
```suggestion
    this.data.stream.set('extra', { 'bbb_options': { 'record': true, 'autoStartRecording': false, 'muteOnStart': true } });
```

File: app/components/forms/events/view/videoroom-form.hbs [hunk 2]
```diff
@@ -112,6 +112,35 @@
...
     </div>
   {{/if}}
+  {{#if (eq this.data.stream.videoChannel.provider 'bbb')}}
+    <div class="field">
+      <label>
+          {{t 'Record'}}
+      </label>
+      <UiCheckbox
+        class="toggle"
+        @checked={{this.data.stream.extra.options.record}}
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.hbs:122]:
```suggestion
        @checked={{this.data.stream.extra.bbb_options.record}}
```

File: app/components/forms/events/view/videoroom-form.hbs [hunk 3]
```diff
@@ -112,6 +112,35 @@
...
+        @onChange={{action (mut this.data.stream.extra.options.record)}} />
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.hbs:123]:
```suggestion
        @onChange={{action (mut this.data.stream.extra.bbb_options.record)}} />
```

File: app/components/forms/events/view/videoroom-form.hbs [hunk 4]
```diff
@@ -112,6 +112,35 @@
...
+    </div>
+    <div class="field">
+      <label>
+          {{t 'Auto Start Recording'}}
+      </label>
+      <UiCheckbox
+        class="toggle"
+        @checked={{this.data.stream.extra.options.autoStartRecording}}
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.hbs:131]:
```suggestion
        @checked={{this.data.stream.extra.bbb_options.autoStartRecording}}
```

File: app/components/forms/events/view/videoroom-form.hbs [hunk 5]
```diff
@@ -112,6 +112,35 @@
...
+        @onChange={{action (mut this.data.stream.extra.options.autoStartRecording)}} />
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.hbs:132]:
```suggestion
        @onChange={{action (mut this.data.stream.extra.bbb_options.autoStartRecording)}} />
```

File: app/components/forms/events/view/videoroom-form.hbs [hunk 6]
```diff
@@ -112,6 +112,35 @@
...
+    </div>
+    <div class="field">
+      <label>
+          {{t 'Mute On Start'}}
+      </label>
+      <UiCheckbox
+        class="toggle"
+        @checked={{this.data.stream.extra.options.muteOnStart}}
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.hbs:140]:
```suggestion
        @checked={{this.data.stream.extra.bbb_options.muteOnStart}}
```

File: app/components/forms/events/view/videoroom-form.hbs [hunk 7]
```diff
@@ -112,6 +112,35 @@
...
+        @onChange={{action (mut this.data.stream.extra.options.muteOnStart)}} />
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.hbs:141]:
```suggestion
        @onChange={{action (mut this.data.stream.extra.bbb_options.muteOnStart)}} />
```

File: app/components/forms/events/view/videoroom-form.js [hunk 8]
```diff
@@ -225,5 +226,8 @@ export default class VideoroomForm extends Component.extend(FormMixin) {
     if (this.data.stream.extra === null && ['vimeo', 'youtube'].includes(this.data.stream.videoChannel.get('provider'))) {
       this.data.stream.set('extra', { 'autoplay': true, 'loop': false });
     }
+    if (!this.data.stream.extra.options && this.data.stream.videoChannel.get('provider') === 'bbb') {
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.js:229]:
```suggestion
    if (!this.data.stream.extra.bbb_options && this.data.stream.videoChannel.get('provider') === 'bbb') {
```

File: app/components/forms/events/view/videoroom-form.js [hunk 9]
```diff
@@ -225,5 +226,8 @@ export default class VideoroomForm extends Component.extend(FormMixin) {
...
+      this.data.stream.set('extra', { 'options': { 'record': true, 'autoStartRecording': false, 'muteOnStart': true } });
```
Comment (iamareebjamal) [inline app/components/forms/events/view/videoroom-form.js:230]:
```suggestion
      this.data.stream.set('extra', { 'bbb_options': { 'record': true, 'autoStartRecording': false, 'muteOnStart': true } });
```

File: app/models/video-stream.ts [hunk 10]
```diff
@@ -6,9 +6,15 @@ import Microlocation from './microlocation';
...
 
+export interface Options {
+  record: boolean;
+  autoStartMeeting: boolean;
+  muteOnStart: boolean;
+}
 export interface Extra {
   autoplay: boolean;
   loop: boolean;
+  options: Options;
```
Comment (iamareebjamal) [inline app/models/video-stream.ts:17]:
```suggestion
  bbb_options: Options;
```

**Pull request merged**